types/query/pagination.go

```diff
@@ -0,0 +1,110 @@
+package query
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store/types"
+)
+
+// defaultLimit is the default `limit` for queries
+// if the `limit` is not supplied, paginate will use `defaultLimit`
+const defaultLimit = 100
+
+// Paginate does pagination of all the results in the PrefixStore based on the
+// provided PageRequest. onResult should be used to do actual unmarshaling.
+//
+// Ex:
+//  func (q BaseKeeper) QuerySome(c context.Context, req *types.QuerySomeRequest)
```

Comment:
I was thinking something more like the following

```
func ExamplePaginate() {
...
```

to be included in `paginatiotion_test.go` alongside other tests (as per https://golang.org/pkg/testing/#hdr-Examples).
It may require some mocking, though that would show up in the documentation and provide users with a working example.

Reply:
Great work on the testable example. Now please let's remove the code example snippet from here, shall we?

Reply:
Thanks, sure will update